smartsim/_core/control/controller.py

```diff
@@ -572,7 +582,11 @@ def _launch_step(
         if completed_job is None and (
             entity.name not in self._jobs.jobs and entity.name not in self._jobs.db_jobs
         ):
-            self.symlink_output_files(job_step, entity)
+            # make sure it's not an _AnonymousBatchJob
```

Comment:
I'm not sure this comment is accurate (since we're looking for `isinstance(Model)`). Consider tweaking your conditional to state your intent instead of relying solely on the comment?

```
not_a_model = not isinstance(entity, Model)
is_non_batch_model = not not_a_model and entity.batch_settings

if not_a_model or is_non_batch_model:
   self.symlink_output_files(job_step, entity)

```

Reply:
if we're REALLY looking to see that it's an `_AnonymousBatchJob` what stops us from checking directly w/`if not isinstance(entity, _AnonymousBatchJob)`?

Reply:
Originally tried checking with `if not isinstance(entity, _AnonymousBatchJob)` and it didn't work, which I thought was interesting. So that's why it's broken up into an ugly conditional.

Reply:
waaaaaait actually i think i tried `if not isinstance(job_step, _AnonymousBatchJob)`. oops! let me try your way!

Reply:
That also doesn't work.

Reply:
I pulled the symlinking calls out of `launch_step` because honestly they didn't need to be there. Because I did that, the AnonymousBatchJob doesn't overwrite its symlink anymore, so this logic is no longer needed!